### PR TITLE
helm: pass in config.backup to backup-flags

### PR DESCRIPTION
**Pull request by derekperkins**
@enisoc

### CODE REVIEW

**Comment by enisoc** [comment]
Now I'm getting `flag provided but not defined: -restore_from_backup`. We need to separate that since it's a vttablet-only flag.

**Comment by derekperkins** [comment]
I just pushed a fix that only uses the `-restore_from_backup` flag on `vttablet`

File: helm/vitess/templates/_vttablet.tpl [hunk 1]
```diff
@@ -328,7 +328,7 @@ spec:
         -orc_api_url "http://orchestrator.{{ $namespace }}/api"
         -orc_discover_interval "5m"
 {{ end }}
-{{ include "backup-flags" $config.backup | indent 8 }}
+{{ include "backup-flags" (tuple $config.backup "vtctld") | indent 8 }}
```
Comment (enisoc) [inline helm/vitess/templates/_vttablet.tpl]: `"vttablet"`

**Comment by derekperkins** [comment]
Good catch on my copy/paste error. I just amended that last commit.

File: helm/vitess/templates/_helpers.tpl [hunk 2]
```diff
@@ -105,21 +105,31 @@ export EXTRA_MY_CNF="$FLAVOR_MYCNF:/vtdataroot/tabletdata/report-host.cnf:/vt/co
 # backup flags - expects config.backup
 #############################
 {{- define "backup-flags" -}}
+# set tuple values to more recognizable variables
```
Comment (enisoc) [inline helm/vitess/templates/_helpers.tpl]: The flag comment strikes again. This is getting injected into the command line, hiding everything after.

Comment (derekperkins) [reply helm/vitess/templates/_helpers.tpl]: I hate the flag comment!

**Comment by derekperkins** [comment]
Flag comment destroyed

**Comment by enisoc** [comment]
LGTM

**Pull request merged**